### PR TITLE
Issue #6129: Missing node_preview()

### LINKED ISSUE #6129: [DX] Restore theme_node_preview() for theming and compatibility
I know that this sounds like a task for adding documentation and a change record, so better filed under https://github.com/backdrop-ops/docs.backdropcms.org/issues, however, there may be core-related things that we need to do, such as adding a wrapper function with a `@deprecated` and a watchdog message thrown etc.

This came up in [a thread in Zulip](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/D7.20theme_node_preview.20becomes.20node_preview.3F):
> Hi, in D7 we used `theme_node_preview` to modify how a node is previewed before saving, and I'm trying to get the same result in Backdrop, but it seems that in Backdrop there's a different approach to doing this. But I can't understand what it is! It looks like I should use `node_preview`. Can someone explain how this would be done?

Here's what we know:
- We've removed node previews completely in Backdrop v1.0.0 with #218 and https://github.com/backdrop/backdrop/pull/601
- We then re-introduced node previews much more improved in v1.11.0, with #3062 and https://github.com/backdrop/backdrop/pull/2290
- We iterated after that, with a series of follow-up bug fixes and UX improvements:
  - #3275
  - #3336
  - #3338
  - #3361
  - #4856
  - #5018
  - #5067

Clues: In the original PR that reintroduced the node previews, there's a `@see node_form_build_preview()` in the docblock of `node_preview()`, however that eventually has changed to `@see node_form_preview()` later. I cannot see any theme function being called in that function, so not sure how people can override it 🤷🏼

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/6129

### CODE REVIEW

File: core/modules/node/node.theme.inc [hunk 1]
```diff
@@ -32,6 +32,22 @@ function theme_node_preview_banner_form($variables) {
   return $output;
 }
 
+/**
+ * Returns HTML for node_preview().
+ *
+ * @param $variables
```
Comment (herbdool) [inline core/modules/node/node.theme.inc]: @docwilmot you'll need to fix this and the other things phpcs picked up.

Comment (herbdool) [reply core/modules/node/node.theme.inc]: You'll also need to add a `@return array` here I think.

File: core/modules/node/node.theme.inc [hunk 2]
```diff
@@ -32,6 +32,22 @@ function theme_node_preview_banner_form($variables) {
   return $output;
 }
 
+/**
+ * Returns HTML for node_preview().
```
Comment (avpaderno) [inline core/modules/node/node.theme.inc]: Would not *Returns HTML for node previews.* be preferable?

Comment (klonos) [reply core/modules/node/node.theme.inc]: FTR, the description in the respective function in D7 has the following:
> Returns HTML for a node preview for display during node creation and editing.

See https://api.drupal.org/api/drupal/modules%21node%21node.pages.inc/function/theme_node_preview/7.x

File: core/modules/node/node.theme.inc [hunk 3]
```diff
@@ -32,6 +32,22 @@ function theme_node_preview_banner_form($variables) {
...
+ *
+ * @param $variables
+ *   An associative array containing:
+ *   - node_preview: A render element representing the node preview.
+ *
+ * @see node_preview_banner_form()
```
Comment (klonos) [inline core/modules/node/node.theme.inc:43]: Should we also be pointing to `node_preview()` here?:
```suggestion
 * @see node_preview()
 * @see node_preview_banner_form()
```

File: core/modules/node/node.theme.inc (hunk 1 above)
Comment (docwilmot) [reply core/modules/node/node.theme.inc]: Interestingly, none of the other functions in that file have an @return.

File: core/modules/node/node.theme.inc (hunk 2 above)
Comment (docwilmot) [reply core/modules/node/node.theme.inc]: Went with @klonos suggestion

File: core/modules/node/node.theme.inc [hunk 4]
```diff
@@ -32,6 +32,23 @@ function theme_node_preview_banner_form($variables) {
...
+ * Returns HTML for a node preview for display during node creation and editing.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - node_preview: A render element representing the node preview.
+ *
+ * @see node_preview()
+ * @see node_preview_banner_form()
+ * @see theme_node_preview_banner_form()
+ * @ingroup themeable
```
Comment (quicksketch) [inline core/modules/node/node.theme.inc:46]:
```suggestion
 *
 * @ingroup themeable
 *
 * @since 1.28.0 Function added.
```

**Comment by quicksketch** [comment]
Merged into 1.x for 1.28.0. Thanks @docwilmot, @herbdool, @kiamlaluno, and @klonos!

**Pull request merged**